src/Bridge/Symfony/Routing/Router.php

```diff
@@ -72,7 +74,12 @@ public function match($pathInfo)
         $pathInfo = str_replace($baseContext->getBaseUrl(), '', $pathInfo);
 
         $request = Request::create($pathInfo, 'GET', [], [], [], ['HTTP_HOST' => $baseContext->getHost()]);
-        $context = (new RequestContext())->fromRequest($request);
+        try {
+            $context = (new RequestContext())->fromRequest($request);
+        } catch (RequestExceptionInterface $e) {
+            throw new RouteNotFoundException('Invalid request context.');
```

Comment:
Throwing a `RouteNotFoundException` is weird. According to the PHPDoc, it should be `NoConfigurationException` or `ResourceNotFoundException` or `MethodNotAllowedException`.

Reply:
discussed with @dunglas and we accepted this one because it's the only possibility so that it's catched correctly after. We may change the phpdoc to reflect this though.

FYI this has been merged to 2.4 but several people asked for a backport.

Reply:
`Symfony\Component\Routing\Exception\ResourceNotFoundException` implements the `Symfony\Component\Routing\Exception\ExceptionInterface`. So it's okay: the exception will be caught by the `IriConverter::getItemFromIri()` method!